pkg/clusterversion/README.md

```diff
@@ -0,0 +1,249 @@
+        version.txt = v24.1.0-beta.1")
+  
+  rel1 --> rel2
+  
+  rel2("R.2: Mint release
+        Latest = 24.1
+        MinSupported = 23.1
+        PreviousRelease = 23.2
+        developmentBranch = false
+        finalVersion = V24_1
+        version.txt = v24.1.0")
+
+
+  base -- master --> master1
+  
+  master1("M.1: Bump min supported
+           Latest = 23.2-x
+           MinSupported = 23.2
+           PreviousRelease = 23.2
+           version.txt = v24.1.0-alpha.00000000")
+            
+  master1 --> master2
+    
+  master2("M.2: Bump current version
+           Latest = 24.1-2
+           MinSupported = 23.2
+           PreviousRelease = 23.2
+           version.txt = v24.2.0-alpha.00000000")
+            
+  master2 --> master3
+            
+  master3("M.3: Bump previous release
+           Latest = 24.1-x
+           MinSupported = 23.2
+           PreviousRelease = 24.1
+           version.txt = v24.2.0-alpha.00000000")
+            
+  master3 --> master4
+
+  master4(M.4: Finalize gates and bootstrap data for released version)
+```
+
+## Release branch changes
+
+The changes below happen directly on the release branch (and only on that
+branch). They are not normal backports of changes from `master`.
+
+### R.1: Prepare for beta
+
+**What**: This change prepares the branch for releasing the initial beta. The
+first beta is special because starting with this version we allow production
+clusters from the previous version to be upgraded to the beta version; and,
+consequently, we allow and support upgrading from the beta cluster version to
+the final release cluster version. This transition is achieved by setting
+`developmentBranch` to `false`; see `developmentBranch` and `DevOffset` in the
+code for more details.
+
+**When**: When we are ready to select the first beta candidate.
+
+**Checklist**:
+ - [ ] Set `developmentBranch` constant to `false`
+ - [ ] Update `version.txt` to the beta version, e.g. `24.1.0-beta.1`
+ - [ ] Regenerate docs (`./dev gen docs`)
+ - [ ] Regenerate expected test data results as needed
+
+**Example PR:** [#113912](https://github.com/cockroachdb/cockroach/pull/113912)
+
+### R.2: Mint release
+
+**What**: This change finalizes the cluster version for the release.
+
+**When**: When we are ready to release the final .0 version. (TODO: or RC1?)
+
+**Checklist**:
+- [ ] Replace temporary constant for current release (e.g. `V24_1`) with a
+  cluster version key, associated with a "final" (`Internal=0`) version (e.g.
+  `24.1`)
+- [ ] Set `finalVersion` constant to the key (e.g. `V24_1`)
+- [ ] Update `pkg/build/version.txt` to the beta version (e.g. `24.1.0-beta.1`)
+- [ ] Regenerate docs (`./dev gen docs`)
+- [ ] Regenerate expected test data results as needed
+
+**Example PR:** [#112347](https://github.com/cockroachdb/cockroach/pull/112347)
+
+## Master branch changes
+
+The changes below happen on the `master` branch after forking the release, and
+are not backported to the release branch.
```

Comment:
Maybe explicitly say `release-major.minor`, so we don't mix it with the do-zero branch?